src/runtime/pyscope.cs

```diff
@@ -0,0 +1,427 @@
+        }
+
+        public void ImportScope(PyScope scope)
+        {
+            int result = Runtime.PyDict_Update(variables, scope.variables);
+            if (result < 0)
+            {
+                throw new PythonException();
+            }
+            InitVariables();
+        }
+
+        /// <summary>
+        /// ImportModule Method
+        /// </summary>
+        /// <remarks>
+        /// The import .. as .. statement in Python.
+        /// Import a module ,add it to the variables dict and return the resulting module object as a PyObject.
+        /// </remarks>
+        public PyObject ImportModule(string name)
+        {
+            return ImportModule(name, name);
+        }
+
+        /// <summary>
+        /// ImportModule Method
+        /// </summary>
+        /// <remarks>
+        /// The import .. as .. statement in Python.
+        /// Import a module ,add it to the variables dict and return the resulting module object as a PyObject.
+        /// </remarks>
+        public PyObject ImportModule(string name, string asname)
+        {
+            Check();
+            PyObject module = PythonEngine.ImportModule(name);
+            if (asname == null)
+            {
+                asname = name;
+            }
+            SetVariable(asname, module);
+            return module;
+        }
+
+        /// <summary>
+        /// Execute method
+        /// </summary>
+        /// <remarks>
+        /// Execute a Python ast and return the result as a PyObject.
+        /// The ast can be either an expression or stmts.
+        /// </remarks>
+        public PyObject Execute(PyObject script)
+        {
+            Check();
+            IntPtr ptr = Runtime.PyEval_EvalCode(script.Handle, variables, variables);
+            Runtime.CheckExceptionOccurred();
+            if (ptr == Runtime.PyNone)
+            {
+                Runtime.XDecref(ptr);
+                return null;
+            }
+            return new PyObject(ptr);
+        }
+
+        public T Execute<T>(PyObject script)
+        {
+            Check();
+            PyObject pyObj = Execute(script);
+            if (pyObj == null)
+            {
+                return default(T);
+            }
+            var obj = (T)ToManagedObject<T>(pyObj);
+            return obj;
+        }
+
+        public T ExecuteVariable<T>(string name)
+        {
+            PyObject script = GetVariable(name);
+            return Execute<T>(script);
+        }
+
+        /// <summary>
+        /// Eval method
+        /// </summary>
+        /// <remarks>
+        /// Evaluate a Python expression and return the result as a PyObject
+        /// or null if an exception is raised.
+        /// </remarks>
+        public PyObject Eval(string code)
+        {
+            Check();
+            var flag = (IntPtr)Runtime.Py_eval_input;
+            IntPtr ptr = Runtime.PyRun_String(
+                code, flag, variables, variables
+            );
```

Comment:
There is already code for evaluating string inputs, you shouldn't call `Runtime.` functions here.